tests/phpunit/integration/Core/Util/Google_URL_Matcher_TraitTest.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * Google_URL_Matcher_TraitTest
+ *
+ * @package   Google\Site_Kit\Tests\Core\Util
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\Util;
+
+use Google\Site_Kit\Tests\Core\Util\Google_URL_Matcher;
+use Google\Site_Kit\Tests\TestCase;
+
+/**
+ * @group Util
+ */
+class Google_URL_Matcher_TraitTest extends TestCase {
+
+	private $matcher;
+
+	public function setUp() {
+		parent::setUp();
+
+		$this->matcher = new Google_URL_Matcher();
+	}
+
+	/**
+	 * @dataProvider data_is_url_match
+	 *
+	 * @param string $url      The URL.
+	 * @param string $compare  The URL to compare.
+	 * @param bool   $expected Whether a match is expected or not.
+	 */
+	public function test_is_url_match( $url, $compare, $expected ) {
+		$result = $this->matcher->is_url_match( $url, $compare );
+		if ( $expected ) {
+			$this->assertTrue( $result );
+		} else {
+			$this->assertFalse( $result );
+		}
```

Comment:
This could use `assertSame( $expected, $result )` which is more strict than `assertEquals` instead of separate methods here.

Reply:
Potentially, but since this is just a true/false comparison, I personally prefer to use the assertions intended for that specifically.